p4-16/psa/PSA.mdk

```diff
@@ -2069,7 +2069,16 @@ specified as a parameter when instantiating the action selector.
 Action selector members may only specify action types defined
 in the `actions` attribute of the implemented table. All actions in a group must be
 of the same type. The action parameters for actions in the same group are allowed
-to differ, and the action of different groups in a selector may be different.
+to differ, and the action of different groups in a selector may be different.  More precisely, a PSA implementation is allowed to, but need not, support an action selector group where different members use different action types.
```

Comment:
Line wrap?
Doesn't the precision contradict the statement above:

> All actions in a group must be of the same type.

Reply:
If the way it is written now is confusing, I would prefer to modify the sentence "All actions in a group must be of the same type."  Except in rare circumstances where we actually want to prevent something, I think PSA should specify "implementations should support _at least this_", and should usually not say "an implementation _cannot_ do X".

Reply:
I think the first sentence should be changed. Right now IMO they strictly contradict each other.

Reply:
Eliminated the contradiction, by changing the whole paragraph.  Hopefully the meaning is still reasonably clear.